Userland/Libraries/LibCompress/Brotli.cpp

```diff
@@ -0,0 +1,1149 @@
+/*
+ * Copyright (c) 2022, Michiel Visser <opensource@webmichiel.nl>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/BinarySearch.h>
+#include <AK/MemoryStream.h>
+#include <AK/QuickSort.h>
+#include <LibCompress/Brotli.h>
+#include <LibCompress/BrotliDictionary.h>
+
+namespace Compress {
+
+Optional<size_t> BrotliCanonicalCode::read_symbol(InputBitStream& input_bit_stream)
```

Comment:
I'm guessing that the input bit stream is little endian? Then it would be great to not have an old stream for new code, but use a reimplementation `LittleEndianInputBitStream` with `Core::Stream`. This will be super convenient for porting other bit stream users!